app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -61,48 +61,30 @@ class SchoolDisciplineDashboard extends React.Component {
       title: "Incidents by " + selectedChart};
   }
 
-  sortChartData(chart) {
-    switch(chart.type) {
-      case 'time': return this.sortByTime(chart.disciplineIncidents);
-      case 'day': return this.sortByDay(chart.disciplineIncidents);
-      default: return chart.disciplineIncidents;
+  sortChartKeys(chartKeys) {
+    switch(this.state.selectedChart) {
+    case 'time': return this.sortedTimes(chartKeys);
+    case 'day': return this.sortedDays(chartKeys);
+    case 'grade': return this.sortedGrades(chartKeys);
+    default: return chartKeys;
     }
   }
 
-  sortByDay(incidentsGroupedByDay) {
-    const dayOrder = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
-    let sortedHash = {};
-    dayOrder.forEach((day) => {
-      if (incidentsGroupedByDay[day]) sortedHash[day] = incidentsGroupedByDay[day];
-    });
-    return sortedHash;
+  sortedDays(chartKeys) {
+    return ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'].filter((day) => chartKeys.indexOf(day) >= 0);
   }
 
-  sortByTime(incidentsGroupedByTime) {
-    const sortedTimes = Object.keys(incidentsGroupedByTime).sort((a, b) => {
-      // Times are coming in as '5 AM', '4 PM', or 'Not Logged'
-
-      if (a === b) return 0;
-      if (a === 'Not Logged') return -1;
-      if (b === 'Not Logged') return 1;
-
-      let a_Hour = parseInt(a.split(' ')[0]);
-      let a_AmPm = a.split(' ')[1];
-      let b_Hour = parseInt(b.split(' ')[0]);
-      let b_AmPm = b.split(' ')[1];
-
-      if (a_AmPm === 'AM' && b_AmPm == 'PM') return -1;
-
-      if (a_AmPm === 'PM' && b_AmPm == 'AM') return 1
-
-      return a_Hour - b_Hour;
+  sortedTimes(chartKeys) {
+    return chartKeys.sort((a, b) => {
+      if (a == "Not Logged") return -1;
+      if (b == "Not Logged") return 1;
+      return new Date('1970/01/01 ' + a) - new Date('1970/01/01 ' + b);
     });
+  }
+
+  sortedGrades(chartKeys) {
+    return ['PK', 'KF', '1','2','3','4','5','6','7','8','9','10','11','12'].filter((grade) => chartKeys.indexOf(grade) >= 0);
```

Comment:
Same as with days of the week above, I don't think we want to be filtering out grades with no incidents. If 3rd grade has no incidents, that could be a really meaningful thing for an educator to see. Of course some schools will have no students in certain grades, like an elementary school that has no students in 9 through 12. But I think we should handle that by looking directly at the student data, not by asking if there are any incidents at a grade level, since zero incidents could occur at a valid grade level.